JASP-Engine/JASP/R/contingencytables.R

```diff
@@ -15,1778 +15,1412 @@
-
-		if (options$percentagesColumn) {
-
-			if (class(col.proportions.matrix[1,1]) == "character") {
-				col.proportions <- col.proportions.matrix[1,]
-			} else {
-				colproportion <- apply(col.proportions.matrix, 2, base::sum)
-			}
-
-			col.proportions <- as.list(colproportion)
-			col.proportions <- .clean(col.proportions)
-			names(col.proportions) <- paste(names(col.proportions),"[col.proportions]", sep="")
-
-			if (class(row.proportions.matrix[1,1]) == "character") {
-				col.proportions[["total[col.proportions]"]] <- ""
-			} else {
-				row.sum <- base::margin.table(counts.matrix, 1)
-				row.prop <- base::prop.table(row.sum)
-				col.proportions[["total[col.proportions]"]] <- .clean(base::sum(row.prop))
-			}
-
-			col.proportions<-c(row.col.proportions, col.proportions)
-
-			row <- c(row,  col.proportions)
-		}
-
-
-		if (options$percentagesTotal) {
-
-			if (class(proportions.matrix[1,1]) == "character") {
-				proportions <- proportions.matrix[1,]
-			} else {
-				proportions <- apply(proportions.matrix, 2, base::sum)
-			}
-
-			proportions <- as.list(proportions)
-			proportions <- .clean(proportions)
-			names(proportions) <- paste(names(proportions),"[proportions]", sep="")
-
-			if (class(proportions.matrix[1,1]) == "character") {
-				proportions[["total[proportions]"]] <- ""
-			} else {
-				proportions[["total[proportions]"]] <- .clean(base::sum(proportions.matrix))
-			}
-
-			proportions <- c(row.proportions, proportions)
-
-			row <- c(row,  proportions)
-		}
-
-	} else {
-
-		row <- list()
-	}
-
-	row[[var.name]] <- "Total"
-	if (options$countsExpected == FALSE && options$percentagesRow == FALSE && options$percentagesCol == FALSE && options$percentagesTotal == FALSE)
-		row[[".isNewGroup"]] <- TRUE
-
-	for (layer in names(group)) {
-
-		level <- group[[layer]]
-
-		if (level == "") {
-
-			row[[layer]] <- "Total"
-
-		} else {
-
-			row[[layer]] <- level
-		}
-	}
-
-	rows[[length(rows)+1]] <- row
-
-	rows
+# Combinations of rows, columns, layers ----
+.crossTabComputeAnalyses <- function(dataset, options, ready) {
+  rows    <- as.vector(options$rows,    "character")
+  columns <- as.vector(options$columns, "character")
+  
+  if (length(rows) == 0)
+    rows <- ""
+  
+  if (length(columns) == 0)
+    columns <- ""
+  analyses <- list()
+  analyses$analyses <- data.frame("columns" = columns, stringsAsFactors = FALSE)
+  analyses$analyses <- cbind(analyses$analyses, 
+                             "rows" = rep(rows, each = dim(analyses$analyses)[1]), 
+                    stringsAsFactors = FALSE)
+  
+  for (layer in options$layers) {
+    layer.vars <- as.vector(layer$variables, "character")
+    analyses$analyses <- cbind(analyses$analyses, 
+                               rep(layer.vars, each = dim(analyses$analyses)[1]), 
+                      stringsAsFactors = FALSE)
+    names(analyses$analyses)[dim(analyses$analyses)[2]] <- layer$name
+  }
+  
+  analyses$analyses <- .dataFrameToRowList(analyses$analyses)
```

Comment:
For the `analyses `and the `groups`, each subpart (`analysis` and `group`) needs to be a list. 
I thought it'd be better to keep the old way and create lists of lists rather than a making a list per row of the data.frame inside the `for` loops